src/providers/openExchangeRates.ts

```diff
@@ -40,7 +40,7 @@ const openExchangeRatesRateMap = createReducedRateMap(
 
 const query = async (date: string, codes: string[]): Promise<NewRates> => {
   const rates = { [date]: {} }
-  if (codes.length === 0) return rates
+  if (codes.length === 0 || codes.every(code => code === 'USD')) return rates
```

Comment:
'Every' will always loop over the entire array.
But if you did 'codes.find(code => code !== 'USD') == null' then it would only loop over the entire array in case everything is 'USD'.